### ISSUE TITLE
mintclub.network("avalanchefuji").token('SuperSquad').create({...values}) returns undefined

### ISSUE THREAD

**Issue by Resister-boy** [Steemhunt/mint.club-v2-sdk#1]
I just follow the official documentation but function create({...value}) returns undefined. We are planning build with mintclub-sdk how can i get returns Promise<TransactionReceipt> from function create({})

**Comment by ggomaeng** [comment]
could you please share the code you used?

**Comment by Resister-boy** [comment]
```javascript
  const mmValue = mintclub.network("avalanchefuji").nft('SuperSquad');

  const createToken = async () => {
    try {
      const res = await mmValue.create({
        name: 'SuperSquad',

        reserveToken: {
          address: '0x4200000000000000000000000000000000000006',
          decimals: 18,
        },
        // Bonding curve data
        curveData: {
          curveType: 'EXPONENTIAL',

          stepCount: 10, // how granular the curve is
          maxSupply: 10_000, // NFT max supply
          initialMintingPrice: 0.01, // starting price, 0.01 WETH
          finalMintingPrice: 0.1, // ending price, 0.1 WETH
          creatorAllocation: 100, // initial supply to the deployer
        },
        metadataUrl: 'https://api.filebase.io/v1/ipfs',
      });

      console.log('res', res);
    } catch (e) {
      console.log(e);
    }
  };
```

**Comment by ggomaeng** [comment]
Note that create function can also return undefined if it runs into an error.

You should use use the callback functions provided.

The flow goes 
1. Create token function called https://sdk.mint.club/docs/sdk/network/token/create#onsignaturerequest-optional
2. User signs the transaction https://sdk.mint.club/docs/sdk/network/token/create#onsigned-optional
3. The transaction is successfully submitted https://sdk.mint.club/docs/sdk/network/token/create#onsuccess-optional

If any of this fails, it will trigger onError
https://sdk.mint.club/docs/sdk/network/token/create#onerror-optional

----

Also, from the code I can see that an invalid metadataUrl is provided.

Please check the example code [here](https://github.com/Steemhunt/orakle-mintclub/blob/82533a03a3077d1881c658833cef5d254baf4f59/app/create/page.tsx#L134)

**Comment by ggomaeng** [comment]
These will also help
- https://sdk.mint.club/docs/sdk/ipfs/upload
- https://sdk.mint.club/docs/sdk/ipfs/uploadMetadata

You will want to pass the ipfs url passed from `uploadMetadata` to the `metadataUrl` function

**Comment by ggomaeng** [comment]
I added the `debugging` section so this could also help.

https://sdk.mint.club/docs/sdk/bonus/debugging

**Comment by Resister-boy** [comment]
Thank you, I will try this all. But there is another weird thing, when I choose 'base' for network, I work successfully while other case(means all chains except 'base') it does not work and returns undefiend for function create. I am wondering is there something difference.

**Comment by ggomaeng** [comment]
> Thank you, I will try this all. But there is another weird thing, when I choose 'base' for network, I work successfully while other case(means all chains except 'base') it does not work and returns undefiend for function create. I am wondering is there something difference.

Do you have enough gas fees to cover the transaction?

**Comment by Resister-boy** [comment]
Yes, Of course. In most case, It does not request deploy token contract with out base.

**Comment by ggomaeng** [comment]
> Yes, Of course. In most case, It does not request deploy token contract with out base.

could you share the code?

**Comment by Resister-boy** [comment]
Sorry, It is same code, there is a only defference "avalanchefuji" to "base"

**Comment by ggomaeng** [comment]
did you try console logging in the onError function?

**Comment by Resister-boy** [comment]
Sorry, I never tried yet. I'll try it now

**Comment by ggomaeng** [comment]
> Sorry, I never tried yet. I'll try it now

did you figure it out?
